eslint.config.cjs

```diff
@@ -1,50 +1,194 @@
+const tsParser = require('@typescript-eslint/parser')
+// const chaiPlugin = require("eslint-plugin-chai-friendly")
+
+//-----------------------------------------------------------------------------
+// ESLint configuration
 module.exports = [
-  js.configs.recommended, // Recommended JS rules
+  // Core ESLint settings
+  {
+    languageOptions: {
+      ecmaVersion: "latest",
+      sourceType: "module",
+    },
+  },
+  eslint.configs.recommended, // Recommended JS rules
   
   {
     files: ["**/*.ts", "**/*.tsx"],
     languageOptions: {
       parser: tsParser,
     },
     plugins: {
-      "@typescript-eslint": ts,
+      "@typescript-eslint": tsPlugin,
+      '@stylistic': stylistic,
+      "n": nodePlugin
+      // "chai-friendly": chaiPLugin
     },
+    ignores: [
+      '/node_modules',  // Ignore dependencies
+      '/lib',           // Ignore build output
+    ],
     rules: {
+      "@typescript-eslint/no-explicit-any": "error", // Disallow 'any'
       "@typescript-eslint/no-unused-vars": "off",
-      "semi": ["warn", "never"],
-      "unicorn/filename-case": "off",
-      "unicorn/prefer-node-protocol": "off",
-      "unicorn/numeric-separators-style": "off",
-      "unicorn/no-hex-escape": "off",
-      "unicorn/better-regex": "off",
-      "unicorn/no-zero-fractions": "off",
-      "unicorn/no-array-for-each": "off",
-      "unicorn/explicit-length-check": "off",
-      "unicorn/no-process-exit": "off",
-      "unicorn/prefer-json-parse-buffer": "off",
-      "no-process-exit": "off",
-      "no-await-in-loop": "off",
-      "no-control-regex": "off",
+      "@typescript-eslint/typedef": "error",
+      
+      // stylistic rules
+      "@stylistic/array-bracket-spacing": ["error", "never"],
+      "@stylistic/array-bracket-newline": ["warn", { "multiline": true }],
+      "@stylistic/indent": ["warn", 2, { "SwitchCase": 1 }],
+      "@stylistic/max-statements-per-line": ["warn", { "max": 1, "ignoredNodes": ['BreakStatement'] }],
+      "@stylistic/no-multi-spaces": "warn", // Disallow multiple spaces except for alignment
+      "@stylistic/no-trailing-spaces": "warn",
+      "@stylistic/object-curly-spacing": ["warn", "never"],
+      "@stylistic/quotes": ["error", "single", { "avoidEscape": true }], // Allows double quotes (") when escaping single quotes
+      "@stylistic/semi": ["warn", "never"],
+
+      // eslint rules
+      "camelcase": "off",
+      "complexity": ["warn", 30],
+      "max-nested-callbacks": "warn",
+
+      "no-control-regex": "warn",
+      "no-console": "off",
+      "no-constant-condition": "warn",
       "no-undef": "off",
       "no-unused-vars": "off",
-      "no-console": "off",
-      "no-unused-expressions": "off",     
-      "no-constant-condition": "off",       
-      "max-nested-callbacks": "off",
-      "camelcase": "off",
-      "node/no-missing-import": "off",
-      "complexity": "off",
-      "indent": ["error", 2, { "SwitchCase": 1 }],
+      "no-unused-expressions": "error",
```

Comment:
For my own understanding, what is the purpose of having both
"@typescript-eslint/no-unused-vars": "off",
and
"no-unused-vars": "off",

Reply:
good point, let me check

Reply:
don't need it, removing